src/Databases/DatabaseAtomic.cpp

```diff
@@ -495,12 +503,17 @@ void DatabaseAtomic::stopLoading()
     DatabaseOrdinary::stopLoading();
 }
 
-void DatabaseAtomic::tryCreateSymlink(const String & table_name, const String & actual_data_path, bool if_data_path_exist)
+void DatabaseAtomic::tryCreateSymlink(const StoragePtr & table, bool if_data_path_exist)
 {
     try
     {
+        String table_name = table->getStorageID().getTableName();
+
+        if (!table->storesDataOnDisk())
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Table {} doesn't have data path to create symlink", table_name);
+
         String link = path_to_table_symlinks + escapeForFileName(table_name);
-        fs::path data = fs::canonical(getContext()->getPath()) / actual_data_path;
```

Comment:
Maybe they shouldn't be created in case of multi-disk configuration at all? Since it will point only to part of the data

Reply:
Often, you create a multi-disk configuration where all the data is written to a single disk and only have other disks listed to allow them to be used when you manually move data parts. Creating sym links in one case and not creating in another similar case could be confusing.

Reply:
That's true, but sometimes you can have multiple disks on one volume, parts will be distributed between multiple disks in "round-robin" fashion, and for this case it will be strange to have symlink only to part of the data.

But this is all minor I guess, and I'm ok with any solution.